runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnContextFactory.java

```diff
@@ -496,4 +501,51 @@ public Timers timers() {
       return timers;
     }
   }
+
+  private static <W extends BoundedWindow> StateContext<W> stateContextFromComponents(
```

Comment:
They are only used by ReduceFnContextFactory (that's why I made them private to this class), and they in any case belong more into runners-core than into SDK - especially now that this method takes SideInputReader, which I would like to move into runners-core.